### PR TITLE
Raise an error if a mandatory parameter is missing in the handler configuration.

### PR DESCRIPTION
**Pull request by raphaelhetzel**

### CODE REVIEW

**Comment by Goltergaul** [comment]
the intension here is that all options must be configured even if they are not uses to have a better overview?

**Comment by andreaseger** [comment]
@Goltergaul no the intention is to raise a NoConfigAvailableError if a param is accessed but it wasn't configured.

For example you access `fieldsets_params` but never configured fieldsets

File: lib/request_handler/base.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@
 require 'request_handler/helper'
 require 'confstruct'
 module RequestHandler
-  class Base
+  class Base # rubocop:disable Metrics/ClassLength
```
Comment (andreaseger) [inline lib/request_handler/base.rb]: can you please move this into the `.rubocop.yml`

File: lib/request_handler/base.rb [hunk 2]
```diff
@@ -71,9 +71,9 @@ def parse_filter_params
       defaults = fetch_defaults('filter.defaults', {})
       defaults.merge(FilterParser.new(
         params:                params,
-        schema:                config.lookup!('filter.schema'),
-        additional_url_filter: config.lookup!('filter.additional_url_filter'),
-        schema_options:        execute_options(config.lookup!('filter.options'))
+        schema:                lookup!('filter.schema'),
+        additional_url_filter: lookup('filter.additional_url_filter'),
```
Comment (andreaseger) [inline lib/request_handler/base.rb]: is it fine for all optional attributes if they get passed as `nil` into the parsers?

Comment (raphaelhetzel) [inline lib/request_handler/base.rb]: yes, they all have defaults or safe constructors.

Comment (andreaseger) [inline lib/request_handler/base.rb]: I meant that defaults in the parser do not matter much if this call will always pass in `nil`. Have you checked this?

Comment (raphaelhetzel) [inline lib/request_handler/base.rb]: Yes. every non bang lookup has special nil handling afterwards.

**Pull request merged**